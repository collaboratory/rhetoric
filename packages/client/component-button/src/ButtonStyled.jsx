import styled, { withTheme, css } from "styled-components";
import { buttonOverrides } from "@collaboratory/craft-util-styled";

const paddingSizes = {
  "xs": "2px 4px",
  "sm": "4px 8px",
  "md": "8px 16px",
  "lg": "12px 24px",
  "xl": "24px 48px"
};

const fontSizes = {
  "xs": "12px",
  "sm": "14px",
  "md": "16px",
  "lg": "18px",
  "xl": "24px"
};

const heightSizes = {
  "xs": "14px",
  "sm": "16px",
  "md": "18px",
  "lg": "20px",
  "xl": "28px"
};

function size(props) {
  const fontSize = fontSizes[props.size || "md"];
  const paddingSize = paddingSizes[props.size || "md"];
  return css`
    font-size: ${fontSize};
    padding: ${paddingSize};
    line-height: ${heightSizes};
    min-height: ${heightSizes};
  `;
}

export const ButtonStyled = styled.button`
  appearance: none;
  font-family: inherit;
  cursor: pointer;
  border-radius: 4px;
  text-transform: uppercase;
  float: ${props => (props.float ? props.float : "none")};
  display: ${props => (props.block ? "block" : "inline-block")};
  text-shadow: 1px 1px 0 rgba(0, 0, 0, 0.4);
  transition: all 0.5s;
  ${size};.

  &:hover {
    box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.15);
    text-shadow: 1px 1px 1px rgba(0, 0, 0, 0.15);
  }

  ${buttonOverrides};
`;
export default withTheme(ButtonStyled);
