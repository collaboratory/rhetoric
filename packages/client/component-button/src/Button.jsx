import React, { Component } from "react";
import PropTypes from "prop-types";

import ButtonStyled from "./ButtonStyled";
import ButtonLink from "./ButtonLink";

export default class Button extends Component {
  static propTypes = {
    to: PropTypes.string
  };

  render() {
    const { to, ...next } = this.props;
    return to ? <ButtonLink to={to} {...next} /> : <ButtonStyled {...next} />;
  }
}
