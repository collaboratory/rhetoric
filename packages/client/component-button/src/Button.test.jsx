import React from "react";
import { shallow } from "enzyme";

import Button from "./Button";
import { ButtonStyled } from "./ButtonStyled";
import { ThemeProvider } from "styled-components";
import DefaultTheme from "@collaboratory/craft-theme-default";

describe("Button Component", () => {
  it("should render without throwing an error", () => {
    const btn = shallow(<Button />);
    expect(btn).toMatchSnapshot();
  });

  it("should render as a link when the to prop is present", () => {
    const btnLink = shallow(<Button to="/" />);
    expect(btnLink).toMatchSnapshot();
  });
});

describe("ButtonStyled Component", () => {
  it("should properly utilize the float prop", () => {
    expect(
      shallow(<ButtonStyled theme={DefaultTheme} float="left" />)
    ).toMatchSnapshot();
  });

  it("should properly utilize the block prop", () => {
    expect(
      shallow(
        <ThemeProvider theme={DefaultTheme}>
          <ButtonStyled block />
        </ThemeProvider>
      )
    ).toMatchSnapshot();
  });
});
