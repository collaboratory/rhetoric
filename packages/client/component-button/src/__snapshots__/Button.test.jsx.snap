// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render as a link when the to prop is present 1`] = `
<WithTheme(Styled(ButtonLinkFilter))
  to="/"
/>
`;

exports[`Button Component should render without throwing an error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ButtonStyled Component should properly utilize the block prop 1`] = `
<styled.button
  block={true}
/>
`;

exports[`ButtonStyled Component should properly utilize the float prop 1`] = `
<button
  className="sc-bdVaJa dovBCI"
/>
`;
